addons/point_of_sale/models/pos_order.py

```diff
@@ -558,6 +633,19 @@ def refund(self):
             'target': 'current',
         }
 
+    @api.model
+    def remove_from_ui(self, server_ids):
+        """ Remove orders from the frontend PoS application
+
+        Remove orders from the server by id.
+        :param server_ids: list of the id's of orders to remove from the server.
+        :type server_ids: list.
+        :returns: list -- list of db-ids for the removed orders.
+        """
+        orders = self.search([('id', 'in', server_ids),('state', '=', 'draft')])
+        orders.write({'state': 'cancel'})
+        orders.sudo().unlink()
```

Comment:
also remove the orders.lines either ondelete="cascade" on order_id of line, or overide def unlink in order to also unlink the lines